compiler/rustc_typeck/src/check/op.rs

```diff
@@ -549,16 +549,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         is_assign: IsAssign,
         op: hir::BinOp,
     ) -> bool {
-        let source_map = self.tcx.sess.source_map();
-        let remove_borrow_msg = "String concatenation appends the string on the right to the \
-                                 string on the left and may require reallocation. This \
-                                 requires ownership of the string on the left";
-
-        let msg = "`to_owned()` can be used to create an owned `String` \
-                   from a string reference. String concatenation \
-                   appends the string on the right to the string \
-                   on the left and may require reallocation. This \
-                   requires ownership of the string on the left";
+        let str_concat_note = "String concatenation appends the string on the right to the
```

Comment:
```suggestion
        let str_concat_note = "string concatenation appends the string on the right to the
```

nit: we tend to avoid capitalization in diagnostics (with the exception of type names and that sort of thing)